.github/workflows/generate-tidal-api.yml

```diff
@@ -0,0 +1,122 @@
+name: Generate API in the TidalAPI module and create a PR if there are changes
+
+on:
+  workflow_dispatch: # manual triggering of this workflow
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  codegen:
+    runs-on: macos-latest
+    env:
+      API_FOLDER: Sources/TidalAPI/
+      GENERATED_FOLDER_NAME: Generated
+      INPUT_FOLDER_NAME: Config/input
+      SCRIPT_NAME: generate_and_clean.sh
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      # Step 2: Install OpenAPI Generator via Homebrew
+      - name: Install OpenAPI Generator with Homebrew
+        run: |
+          brew install openapi-generator
+
+      - name: Run Code Generation Script
+        run: |
+          cd $API_FOLDER
+          chmod +x $SCRIPT_NAME
+          ./$SCRIPT_NAME
+
+      - name: Check for changes
+        id: check_for_changes
+        run: |
+          cd $API_FOLDER
+          git add . 
+          changes_in_generated=$(git status -s $GENERATED_FOLDER_NAME | sed 's/"//g')
+          echo "Changes in $GENERATED_FOLDER_NAME:"
+          echo "$changes_in_generated"
+
+          changes_in_input=$(git status -s $INPUT_FOLDER_NAME | sed 's/"//g')
+          echo "Changes in $INPUT_FOLDER_NAME:"
+          echo "$changes_in_input"
+
+          if [ -n "$changes_in_generated" ] || [ -n "$changes_in_input" ]; then
+            changes_detected="true"
+          else
+            changes_detected="false"
+          fi
+          echo "Changes detected: $changes_detected"
+
+          {
+            echo 'CHANGES_IN_GENERATED<<EOF'
+            echo "$changes_in_generated"
+            echo 'EOF'
+          } >> $GITHUB_OUTPUT
+
+          {
+            echo 'CHANGES_IN_INPUT<<EOF'
+            echo "$changes_in_input"
+            echo 'EOF'
+          } >> $GITHUB_OUTPUT
+
+          echo "CHANGES_DETECTED=$changes_detected" >> $GITHUB_OUTPUT
+
+      - name: No Changes Found - Say Goodbye
+        if: ${{ steps.check_for_changes.outputs.CHANGES_DETECTED == 'false' }}
+        run: |
+            echo "No changes found. Good bye"
+      
+      - name: Prepare git and define a branch name
+        id: prepare_git_and_define_branch_name
+        if: ${{ steps.check_for_changes.outputs.CHANGES_DETECTED == 'true' }}
+        run: |
+          git config user.email "svc-github-tidal-music-tools@block.xyz"
+          git config user.name "TIDAL Music Tools"
+          timestamp=$(date +"%Y-%m-%d-%H-%M-%S")
+          branch_name="tidal-music-tools/Update-TidalAPI-${timestamp}"
+          echo "BRANCH_NAME=$branch_name" >> "$GITHUB_OUTPUT"
+
+      - name: Commit changes
+        if: ${{ steps.check_for_changes.outputs.CHANGES_DETECTED == 'true' }}
+        run: |
+          git checkout -b "${{steps.prepare_git_and_define_branch_name.outputs.BRANCH_NAME }}"
+          git add .
```

Comment:
Files where already added to staging in a previous step, right?

Reply:
Right 🙂 I added them in order to `git status` provide 

`A MyFolder/MyFile.swift`

instead of

`? MyFolder/MyFile.swift`

I will remove extra adding, thank you!